external-import/mandiant/src/connector/base.py

```diff
@@ -529,112 +513,219 @@ def compute_start_structure(self, start_date):
 
-            import_start_date = (
+            base_structure = self.compute_start_structure(
+                self.mandiant_import_start_date
+            )
+            indicator_structure = self.compute_start_structure(
                 self.mandiant_indicator_import_start_date
-                if collection == "indicators"
-                else self.mandiant_import_start_date
             )
+            state = {
+                "vulnerabilities": base_structure,
+                "indicators": indicator_structure,
+                "campaigns": base_structure,
+                "malwares": base_structure,
+                "reports": base_structure,
+                "actors": base_structure,
+            }
+        self.helper.set_state(state)
+
+    def get_state_value(self, collection_name: str, state_key: str) -> Any:
+        """Using this method to get the value of a specific key in the state collection.
+
+        It allows an external user to reset the state. The error is then handle gracefully to exit the current job.
+
+        """
+        try:
+            return self.helper.get_state()[collection_name][state_key]
+        except (KeyError, TypeError) as err:
+            raise StateError(
+                f"State key {state_key} not found in {collection_name} collection"
+            ) from err
 
-            if collection in collection_with_start_epoch:
-                first_run = (
-                    True
-                    if state[collection][STATE_START]
-                    == Timestamp.from_iso(import_start_date).iso_format
-                    else False
+    def set_state_value(self, collection_name: str, state_key: str, value: Any):
+        """Using this method to set the value of a specific key in the state collection.
+
+        See Also:
+            get_state_value
+        """
+        try:
+            state = self.helper.get_state()
+            state[collection_name][state_key] = value
+            self.helper.set_state(state)
+        except (KeyError, TypeError) as err:
+            raise StateError(
+                f"State key {state_key} not found in {collection_name} collection"
+            ) from err
+
+    def process_message(self):
+        self._init_state()
+        for collection in self.mandiant_collections:
+            try:
+                # Handle interval config
+                work_id = None
+                date_now_value = Timestamp.now().value
+                collection_interval = getattr(self, f"mandiant_{collection}_interval")
+
+                last_run_value = Timestamp.from_iso(
+                    self.get_state_value(
+                        collection_name=collection, state_key=STATE_LAST_RUN
+                    )
+                ).value
+
+                # API types related to simple offset
+                collection_with_offset = ["malwares", "actors", "campaigns"]
+                # Start and End, Offset
+                start_offset = self.get_state_value(
```

Comment:
use of safe get_state_value responsible for handling the Error rather than direct key access